packages/mdc-slider/foundation.js

```diff
@@ -577,146 +594,156 @@ class MDCSliderFoundation extends MDCFoundation {
+      x: STEM_LEFT_XPOS - TOP_NECK_RADIUS + (Math.cos(topNeckTheta) * TOP_NECK_RADIUS) + OFFSET_X,
+      y: TOP_LOBE_RADIUS + (topNeckCenterYPos - (Math.sin(topNeckTheta) * TOP_NECK_RADIUS)) + OFFSET_Y,
     };
     const start = {
-      x: topLobeRadius + (topLobeHorizontal / 2) + offsetX,
-      y: 0 + offsetY,
+      x: TOP_LOBE_RADIUS + (topLobeHorizontal / 2) + OFFSET_X,
+      y: 0 + OFFSET_Y,
     };
     const end = {
-      x: topLobeRadius - (topLobeHorizontal / 2) + offsetX,
-      y: 0 + offsetY,
+      x: TOP_LOBE_RADIUS - (topLobeHorizontal / 2) + OFFSET_X,
+      y: 0 + OFFSET_Y,
     };
 
-    // If there is extra, update the start, end, and pointA X positions for correct path
+    // If there is any extra horizontal distance, update the start, end, and pointA X positions
     if (extraHorizontalWidth > 0) {
       start.x = start.x + extraHorizontalWidthRight;
-      end.x = end.x - extraHorizontalWidthLeft;
       pointA.x = pointA.x + extraHorizontalWidthRight;
+      end.x = end.x - extraHorizontalWidthLeft;
     }
 
-    // When the slider reaches a certain point close to the edges the extra on the side closest to the edge
-    // is negative and the angles of the neck need to be updates to create smooth value label
-    if (extraHorizontalWidthLeft < 0 && extraHorizontalWidth > 0) {
-      const topLobeHorizontalLeft = this.adapter_.isRTL() ? (this.rect_.width - translatePx) * 2 : translatePx * 2;
-
-      const leftTopNeckCornerTheta =
-        Math.acos((topNeckArcWidth - (topLobeHorizontalLeft)/2)/(topLobeRadius+topNeckRadius));
-      const leftTopNeckCornerCenterY = Math.sqrt(Math.pow(topLobeRadius+topNeckRadius, 2) -
-        Math.pow(topNeckArcWidth - (topLobeHorizontalLeft)/2, 2));
-
-      pointI.x = topNeckArcWidth - topNeckRadius + (Math.cos(leftTopNeckCornerTheta) * topNeckRadius) + offsetX;
-      pointI.y =
-        topLobeRadius + (leftTopNeckCornerCenterY - (Math.sin(leftTopNeckCornerTheta) * topNeckRadius)) + offsetY;
-      pointH.y = topLobeRadius + leftTopNeckCornerCenterY + offsetY;
-    }
-    if (extraHorizontalWidthRight < 0 && extraHorizontalWidth > 0) {
-      const topLobeHorizontalRight = this.adapter_.isRTL() ? translatePx * 2 : (this.rect_.width - translatePx) * 2;
-
-      const leftTopNeckCornerTheta =
-        Math.acos((topNeckArcWidth - (topLobeHorizontalRight)/2)/(topLobeRadius+topNeckRadius));
-      const leftTopNeckCornerCenterY = Math.sqrt(Math.pow(topLobeRadius+topNeckRadius, 2) -
-        Math.pow(topNeckArcWidth - (topLobeHorizontalRight)/2, 2));
-
-      pointA.x = rightStemX + (topNeckRadius - (Math.cos(leftTopNeckCornerTheta) * topNeckRadius)) + offsetX;
-      pointA.y =
-        topLobeRadius + (leftTopNeckCornerCenterY - (Math.sin(leftTopNeckCornerTheta) * topNeckRadius)) + offsetY,
-      pointB.y = topLobeRadius + leftTopNeckCornerCenterY + offsetY;
+    // When the slider reaches the edge of the slider, the extra horizontal distance on the side closest to the edge
```

Comment:
I'm wondering whether the end-of-slider case is triggering correctly in all instances it should? On http://localhost:8080/slider.html if I look at the first "large values" slider, it always remains centered, whereas the last one shifts the pointer in an effort to keep the balloon mostly in line with the slider. I would think the latter should happen in both cases.

Reply:
This is actually pretty difficult because the way it's set up now is that first the bubble expands to be as wide as possible without necessitating any extra width at the "neck", second to add that extra width if necessary, and lastly shift the bubble back to be in line with the slider. So this should probably be in a separate PR/issue.

Reply:
Entered #4314.